### PR TITLE
add tests for RustSemsimian struct

### PR DESCRIPTION
**Pull request by iQuxLE**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 pub mod utils;
 pub mod similarity;
+use std::fmt;
+
 
 use similarity::{calculate_max_information_content, calculate_phenomizer_score};
 use utils::{convert_list_of_tuples_to_hashmap, expand_term_using_closure, predicate_set_to_key};
 
-type Predicate = String;
+pub type Predicate = String; // change to pub cause of integration test, can be reverted if needed
```
Comment (justaddcoffee) [inline src/lib.rs:14]: Probably we could make all three of these public?

File: src/similarity.rs [hunk 2]
```diff
@@ -203,34 +210,35 @@ mod tests {
...
         let result2 = calculate_semantic_jaccard_similarity(
             &closure_map,
             "BFO:0000002",
             "BFO:0000003",
             &Some(sco_predicate.clone()),
         );
         println!("{result2}");
-        assert_eq!(result2, 0.5);
-
+        assert_eq!(result2, 1.0 / 3.0);
```
Comment (caufieldjh) [inline src/similarity.rs:236]: Why is this expected to be 0.333 when the similarity is between {"BFO:0000003", "BFO:0000002"} and {"BFO:0000003"} ?

**Comment by caufieldjh** [comment]
Nice! Tests pass.
Let's add the coverage (from #59) to the workflow - I'll try to get this done, time permitting.
Then we can merge so this will be available to the other PRs in progress

**Comment by caufieldjh** [comment]
The basic idea here works - just need to get the coverage reports out of the workflow instance

**Comment by caufieldjh** [comment]
Realized a fatal flaw in my logic: pushing reports directly to the working branch would trigger another QC run.
Going to try using Codecov [like OAK does](https://github.com/INCATools/ontology-access-kit/blob/main/.github/workflows/main.yaml) instead.

File: .github/workflows/qc.yml [hunk 3]
```diff
@@ -28,10 +33,35 @@ jobs:
...
+        cargo install cargo-llvm-cov
+        cargo llvm-cov show-env --export-prefix
+        cargo llvm-cov clean --workspace
+        
+        # The following commands run all rust tests, just like cargo test
+        # The first version emits a report for further parsing
+        # The second provides a plain-text report to the stdout
+        # cargo llvm-cov --lcov --output-path coverage.lcov
+
+        cargo llvm-cov
```
Comment (justaddcoffee) [inline .github/workflows/qc.yml:51]: Is this running cargo llvm-cov three times?

File: src/similarity.rs [hunk 4]
```diff
@@ -154,92 +161,88 @@ fn _mrca_and_score(scores: &HashMap<TermID, f64>) -> (Option<TermID>, f64) {
...
             "BFO:0000002",
             &Some(sco_predicate.clone()),
         );
-        println!("{result}");
-        assert_eq!(result, 2.0 / 3.0);
 
+        println!("{:?}", result);
+        assert_eq!(result, 2.0 / 3.0);
+        
+        // NO Predicate
```
Comment (justaddcoffee) [inline src/similarity.rs:185]: delete these two commented lines?

File: src/similarity.rs (hunk 2 above)
Comment (justaddcoffee) [reply src/similarity.rs:194]: I believe this test is correct - the similarity is between {"BFO:0000003", "BFO:0000002"} and {"BFO:0000003", "BFO:0000004"}

NB that BFO:00000004 is included [here](https://github.com/monarch-initiative/semsimian/blob/a11f6438f9e7a0ae4d75b2d1e23cb0c81e4e33ac/src/test_utils.rs#L31)

**Review by justaddcoffee — APPROVED** [review]

File: .github/workflows/qc.yml (hunk 3 above)
Comment (caufieldjh) [reply .github/workflows/qc.yml:51]: Nope - the first two are calls to subfunctions and the last is the "actual" report generation (though in this case it just prints out without a report file)

**Pull request merged**